### PR TITLE
Avatar tests

### PR DESCRIPTION
**Pull request by behowell**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Add ~~screener,~~ perf, snapshot, and unit tests for the Avatar component.

Refactor some Avatar functionality for testability, and fix a few styling bugs in edge cases found through testing.

### CODE REVIEW

**Comment by ecraig12345** [comment]
Instead of checking in the demo images, it would probably be better to add them to our CDN (then add references in TestImages). Otherwise, especially once we switch to running screener from blob storage like Shift just did for v0, we're likely to run into issues the images not loading in time. I'll message you with info about how to do that.

**Comment by ecraig12345** [comment]
See #15097 for uploading images to the CDN

**Review by ecraig12345 — APPROVED** [review]

**Review by khmakoto — APPROVED** [review]

**Review by xugao — APPROVED** [review]

File: apps/perf-test/package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "lodash": "^4.17.15"
   },
   "dependencies": {
+    "@fluentui/react-avatar": "^0.6.0",
```
Comment (ecraig12345) [inline apps/perf-test/package.json:27]: This should be on the latest version (`^0.6.3`) in all packages. Once you fix it in one package you should be able to run `yarn syncpack:fix` to automatically fix the rest.

Comment (behowell) [reply apps/perf-test/package.json]: Ok. I had reverted the package version to match what was checked into the examples package so I wouldn't be modifying unrelated packages in this PR. But I'll go back and update that other package too.

Comment (ecraig12345) [reply apps/perf-test/package.json]: Hmm if the examples package is different then @khmakoto or I must have had a bad merge at some point. I'll fix that in master, sorry.

Comment (ecraig12345) [reply apps/perf-test/package.json]: It's fixed now. Apologies again about requiring the extra changes. :(

Comment (behowell) [reply apps/perf-test/package.json]: Ok thanks -- just pulled in your change
